### PR TITLE
[JG] OPS-12810 - Add a language query parameter our URL's PEGA redirect to 

### PR DESCRIPTION
**Pull request by DreamcastKnight**

### CODE REVIEW

File: test/controllers/CanPayWithinSixMonthsControllerSpec.scala [hunk 1]
```diff
@@ -111,15 +111,35 @@ class CanPayWithinSixMonthsControllerSpec extends ItSpec with PegaRecreateSessio
         JourneyJsonTemplates.`Obtained Can Pay Within 6 months - no`(Origins.Epaye.Bta)(testCrypto)
       )
 
-      val result = controller.canPayWithinSixMonths(Epaye)(fakeRequest)
+      val result = controller.canPayWithinSixMonths(Epaye, None)(fakeRequest)
       testPageIsDisplayed(result, Some(false))
     }
 
+    "change the language cookie to english" in {
```
Comment (AndyHWChung) [inline test/controllers/CanPayWithinSixMonthsControllerSpec.scala:118]: it'd be worth having these sorts of tests for the other two endpoints that do the same thing as well I think so we've got full coverage

File: test/controllers/CanPayWithinSixMonthsControllerSpec.scala [hunk 2]
```diff
@@ -111,15 +111,35 @@ class CanPayWithinSixMonthsControllerSpec extends ItSpec with PegaRecreateSessio
...
+      stubCommonActions()
+      EssttpBackend.CanPayWithinSixMonths.findJourney(testCrypto, Origins.Epaye.Bta)(
+        JourneyJsonTemplates.`Obtained Can Pay Within 6 months - no`(Origins.Epaye.Bta)(testCrypto)
+      )
+
+      val result = controller.canPayWithinSixMonths(Epaye, Some(English))(fakeRequest)
```
Comment (AndyHWChung) [inline test/controllers/CanPayWithinSixMonthsControllerSpec.scala:124]: for a stronger test , we could perhaps call the controller endpoint with a fake request that had welsh in it. That way we test if language actually changes

File: app/controllers/CanPayWithinSixMonthsController.scala [hunk 3]
```diff
@@ -62,8 +64,12 @@ class CanPayWithinSixMonthsController @Inject() (
               val previousAnswers = existingAnswersInJourney(request.journey)
               val form = previousAnswers.fold(CanPayWithinSixMonthsForm.form)(value =>
                 CanPayWithinSixMonthsForm.form.fill(CanPayWithinSixMonthsFormValue.canPayWithinSixMonthsToFormValue(value)))
-
-              Ok(views.canPayWithinSixMonthsPage(form, remainingAmountToPay(j)))
+              val result = Ok(views.canPayWithinSixMonthsPage(form, remainingAmountToPay(j)))
+              lang match {
+                case Some(Welsh)   => result.withCookies(Cookie("PLAY_LANG", "cy"))
+                case Some(English) => result.withCookies(Cookie("PLAY_LANG", "en"))
+                case _             => result
```
[the remark this answers is not in the document]
Comment (DreamcastKnight) [reply app/controllers/CanPayWithinSixMonthsController.scala]: oh sorry you said in other similar places, adding in other spots now

File: test/controllers/CanPayWithinSixMonthsControllerSpec.scala [hunk 4]
```diff
@@ -111,15 +111,35 @@ class CanPayWithinSixMonthsControllerSpec extends ItSpec with PegaRecreateSessio
...
+      cookies(result).get("PLAY_LANG").map(_.value) shouldBe Some("en")
+    }
+
+    "change the language cookie to welsh" in {
```
Comment (AndyHWChung) [inline test/controllers/CanPayWithinSixMonthsControllerSpec.scala:128]: we could check that the language doesn't change if `None` is passed in as well perhaps like:
```scala
Languages.values.foreach{ lang => // there's may or may not be a values you can use
  val result = controller.endpoint(regime, fakeRequestWithLang(lang))
  cookies(result).get("blah").map(_.value) shouldBe Some(lang.code)
}
```

**Review by AndyHWChung — APPROVED** [review]

**Review by Kntanos — APPROVED** [review]

**Pull request merged**